.travis.yml

```diff
@@ -31,30 +51,44 @@ addons:
       - openssh-client
       - openssh-server
       - libopenmpi-dev
+      - clang
+      - clang-tidy-5.0
+      - cmake-data
+      - graphviz
+  homebrew:
+    packages:
+      - gcc@7
```

Comment:
this is a very high version of gcc requirement @hcho3 are we just requiring gcc 5 in xgb?

Reply:
ported from xgb master https://github.com/dmlc/xgboost/blob/master/.travis.yml#L71